vsphere/resource_vsphere_vapp_container_test.go

```diff
@@ -0,0 +1,948 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceVSphereVAppContainerConfigVmSdrsNoVApp(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccResourceVSphereVAppContainerCheckExists(true),
+					testAccResourceVSphereVAppContainerContainsVM("vm"),
+				),
+			},
+			{
+				Config: testAccResourceVSphereVAppContainerConfigVmSdrs(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccResourceVSphereVAppContainerCheckExists(true),
+					testAccResourceVSphereVAppContainerContainsVM("vm"),
+				),
+			},
+		},
+	})
+}
+
+func testAccResourceVSphereVAppContainerPreCheck(t *testing.T) {
+	if os.Getenv("VSPHERE_DATACENTER") == "" {
+		t.Skip("set VSPHERE_DATACENTER to run vsphere_vapp_container acceptance tests")
+	}
+	if os.Getenv("VSPHERE_CLUSTER") == "" {
+		t.Skip("set VSPHERE_CLUSTER to run vsphere_vapp_container acceptance tests")
+	}
+	if os.Getenv("VSPHERE_NETWORK_LABEL_PXE") == "" {
+		t.Skip("set VSPHERE_NETWORK_LABEL_PXE to run vsphere_vapp_container acceptance tests")
+	}
+	if os.Getenv("VSPHERE_DATASTORE") == "" {
+		t.Skip("set VSPHERE_DATASTORE to run vsphere_vapp_container acceptance tests")
+	}
+	if os.Getenv("VSPHERE_NFS_PATH") == "" {
+		t.Skip("set VSPHERE_NFS_PATH to run vsphere_vapp_container acceptance tests")
+	}
+	if os.Getenv("VSPHERE_TEMPLATE") == "" {
+		t.Skip("set VSPHERE_TEMPLATE to run vsphere_vapp_container acceptance tests")
+	}
+}
+
+func testAccResourceVSphereVAppContainerCheckExists(expected bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, err := testGetVAppContainer(s, "vapp_container")
+		if err != nil {
+			if viapi.IsManagedObjectNotFoundError(err) && expected == false {
+				// Expected missing
+				return nil
+			}
+			return err
+		}
+		if !expected {
+			return errors.New("expected vapp_container to be missing")
+		}
+		return nil
+	}
+}
+
+func testAccResourceVSphereVAppContainerCheckFolder(expected string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		vc, err := testGetVAppContainer(s, "vapp_container")
+		if err != nil {
+			return err
+		}
+		f, err := testGetFolder(s, expected)
+		if err != nil {
+			return err
+		}
+		vcp, err := vappcontainer.Properties(vc)
+		if err != nil {
+			return err
+		}
+		if *vcp.ParentFolder != f.Reference() {
+			return fmt.Errorf("expected path to be %s, got %s", expected, vcp.ParentFolder)
+		}
+		return nil
+	}
+}
+
+func testAccResourceVSphereVAppContainerContainsVM(vmName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		vm, err := testGetVirtualMachineProperties(s, vmName)
+		if err != nil {
+			return err
+		}
+		vc, err := testGetVAppContainer(s, "vapp_container")
+		if err != nil {
+			return err
+		}
+		if vm.ParentVApp != nil && vm.ParentVApp.Reference() != vc.Reference() {
+			return fmt.Errorf("VM is not a part of vApp container")
+		}
+		return nil
+	}
+}
+
+func testGetVAppContainer(s *terraform.State, resourceName string) (*object.VirtualApp, error) {
```

Comment:
Can this be thrown into `helper_test.go` just in case it's needed by other tests other than vApp container tests?

This is where most of the other test helper code is stored.

Reply:
Done